src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -87,23 +83,16 @@ def _format_zinc_arguments(settings, distribution):
         :param settings: The jvm platform settings from which to extract the arguments.
         :type settings: :class:`JvmPlatformSettings`
         """
-        zinc_args = [
-            "-C-source",
-            f"-C{settings.source_level}",
-            "-C-target",
-            f"-C{settings.target_level}",
-        ]
-        if settings.args:
-            settings_args = settings.args
-            if any("$JAVA_HOME" in a for a in settings.args):
-                logger.debug(
-                    'Substituting "$JAVA_HOME" with "{}" in jvm-platform args.'.format(
-                        distribution.home
-                    )
-                )
-                settings_args = (a.replace("$JAVA_HOME", distribution.home) for a in settings.args)
-            zinc_args.extend(settings_args)
-        return zinc_args
+        args = list(settings.args)
+        # if source, target or release is already specified, then don't generate args for them.
+        if all(a not in ("-C-source", "-C-target", "-C--release") for a in args):
+            args.extend([f"-C{arg}" for arg in settings.javac_source_target_args(distribution)])
```

Comment:
I think you might be able to avoid the `[` `]` here

Reply:
yeah. that's a good catch.